js/speckled_moth.js

```diff
@@ -1,239 +1,267 @@
+    mean = Number(mean) || 0
+    sigma = Number(sigma) || 1
+    var twoPi = 2* 3.141592
+    var thet = twoPi*Math.random()
+    var rad = Math.pow(-2* Math.log(Math.random()), .5)
+    var normalDistribution = rad* Math.sin(thet)
+    return mean + normalDistribution*sigma
+  };
+
+  // Nice port! :)
+  SM.sample = function (array){
+    var index = Math.floor(Math.random()*array.length )
+    return array[index]
+  };
+
+  SM.pad = function (n, width, z) {
+    z = z || '0';
+    n = n + '';
+    return n.length >= width ? n : new Array(width - n.length + 1).join(z) + n;
+  }
+  // a.distanceFrom(b)
+
+  // ============ end utility functions
+
+  // ==== object definitions
+  // this is now a local (sort of private) variable available
+  // to all the functions defined in this clojure. 
+  var Field = function(args){
+    args = args || {}
+    var defaults = {
+      size: 700, // total area of 490000
+      population: 150,
+      geneSpread: 4,
+      clickSpread: 2
+    };
+    args = merge(defaults, args)
+    this.population = args["population"]
+    this.size = args["size"]
+    this.geneSpread = args["geneSpread"]
+    this.clickSpread = args["clickSpread"]
+    this.fieldColor = randomColorVec()
+    this.mothCount = 0
+    this.liveMoths = []
+    this.deadMoths = []
+    this.view = fieldView(this)
+  };
+
+  // we expose Filed to the public as `SM.Field'
+  SM.Field = Field;
+
+  Field.prototype.averageDistance = function(){
+    var groupColor = Vector.Zero(3)
+    for(var index in this.liveMoths){
+      groupColor = groupColor.add(this.liveMoths[index].color)
+    };
+    return groupColor.multiply(1/this.liveMoths.length)
+  };
+
+  Field.prototype.addMoth = function(color){
+    var args = {
+      field: this, 
+      colorVec: color, 
+      idNum: this.mothCount
+    };
+    f0Moth = new Moth(args);
+    // console.log("made new moth", this.mothCount)
+    f0Moth.view.setup()
+    this.liveMoths.push(f0Moth)
+    this.mothCount++
+
+  };
+  Field.prototype.populate = function(){
+    for(i=0; i<this.population; i++){
+      newColor = randomColorVec()
+      this.addMoth( newColor)
     };
   };
-};
-
-Field.prototype.repopulate = function(){
-  console.log("population boost")
-  if (this.liveMoths.length === 1){
-    newColor = randomColorVec()
-    this.addMoth(newColor)
-    console.log("made new moth in repopulate")
-  } else if (this.liveMoths.length === 0){
-    this.populate();
-    console.log("detected extinction event, rebooting")
-  };
-  var deficit = this.population - this.liveMoths.length;
-  var lastKill = Date.now() - this.deadMoths[this.deadMoths.length-1].timeOfDeath
-  if (deficit < 15 && lastKill < 30000){
-    return;
-  };
-  var pairs = [];
-  while(pairs.length < deficit){
-    var pair0 = sample(this.liveMoths)
-    var pair1 = sample(this.liveMoths)
-    while(pair0 === pair1){
-      pair1 = sample(this.liveMoths)
+
+  Field.prototype.depopulate = function(effectedColor){
```

Comment:
There's a lot going on in this function.  If we're refactoring for clarity, I would consider moving some of the more involved logic into separate functions on the prototype, but I'm kinda nitpicky about small functions.  😄